core/config/project_settings.cpp

```diff
@@ -960,7 +960,7 @@ Error ProjectSettings::save_custom(const String &p_path, const CustomMap &p_cust
 		}
 	}
 	// Check for the existence of a csproj file.
-	if (_csproj_exists(get_resource_path())) {
+	if (_csproj_exists(p_path.get_base_dir())) {
```

Comment:
Not sure what was the intention here, but `get_resource_path()` can point to invalid directory and crash.

Reply:
The intention was to get the path to the directory that contains `project.godot` (which is `res://`). Maybe we can just use the literal string `"res://"` here and it would be handled by `DirAccess`.

The `_csproj_exists` method is just meant to be `FileAccess::exists("res://*.csproj")` but `FileAccess::exists` doesn't support globbing.

Reply:
`get_resource_path()` returns the loaded path. With my change, `save_custom` used to save to a directory different than `res://` will look for `.csproj` inside that directory instead of the load path. Would that be correct? (I think it doesn't affect the editor, as `save_custom()` is normally always used with the current path)

Reply:
I feel like we always look for the `.csproj` in the `res://` directory so looking in a different directory wouldn't be correct. Are you saying the `project.godot` file can be in a directory other than `res://`? Doesn't the editor always assume this file to be in the root project directory?

In which cases is `save_custom` used to save to a directory different than `res://`? I see that this method is exposed to scripting, is this the case you are concerned about? If so, I don't think we care about this scenario. This detection is only used to add the "C#" feature tag to be used by the Project Manager, so any usage outside of that doesn't really matter.

Reply:
I checked how we use `save_custom` and it's used only for creating new projects and for saving project file for exported project. So I guess my concerns are irrelevant.